src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -22,32 +23,35 @@ public class DeleteCommand extends Command {
             + "Parameters: INDEX (must be a positive integer)\n"
             + "Example: " + COMMAND_WORD + " 1";
 
-    public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Person: %1$s";
+    public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Person(s):\n";
+    public static final String MESSAGE_FORMAT_SPECIFIER = "%1$s";
 
-    private final Index targetIndex;
+    public final Set<Index> targetIndexSet;
 
-    public DeleteCommand(Index targetIndex) {
-        this.targetIndex = targetIndex;
+    public DeleteCommand(Set<Index> targetIndexSet) {
+        this.targetIndexSet = targetIndexSet;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Person> lastShownList = model.getFilteredPersonList();
-
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        String deleteMessage = MESSAGE_DELETE_PERSON_SUCCESS;
+        for (Index targetIndex : targetIndexSet) {
+            if (targetIndex.getZeroBased() >= lastShownList.size()) {
```

Comment:
Putting the checking condition in the for loop may cause some issues. Consider the case where the user's middle input number is out of bounds in the array. The program would have already deleted several indexes before throwing out an error. This is an incorrect behaviour as the program should void the whole command and should not execute it at all, since it is an invalid command. You should check for any out of bound errors outside the for loop to prevent this issue.

Reply:
Thanks for pointing this out. I think that the case you mentioned shouldn't be a problem. Before the DeleteCommand is created, the DeleteCommandParser would have sorted the targetIndexSet in reverse order. Hence, any out of bounds index would be checked first and an exception would be thrown before the loop reaches the other indexes in the set. This way it is ensured that no deletion occurs if even one of the indexes are invalid.